### ISSUE TITLE
New Rule: implement `markdownlint`'s `MD028`

### ISSUE THREAD

**Issue by tooth-is-silver** [eslint-markdown/eslint-markdown#746]
### Rule details

Add a rule equivalent to markdownlint's MD028.

The rule should report consecutive blockquote blocks separated only by one or more blank lines, because Markdown parsers may interpret them as either one blockquote or separate blockquotes.

Do not provide an autofix because the rule cannot determine whether the user intended to join or separate the blockquotes.

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
<!-- Incorrect: ambiguous blockquotes -->

> First paragraph.

> Second paragraph.

<!-- Correct: one blockquote with two paragraphs -->

> First paragraph.
>
> Second paragraph.

<!-- Correct: two separate blockquotes -->

> First blockquote.

<!-- Separate blockquotes -->

> Second blockquote.
```

### Prior Art

- markdownlint: [`MD028` - Blank line inside blockquote](https://github.com/DavidAnson/markdownlint/blob/main/doc/md028.md)
